Sources/Action.swift

```diff
@@ -170,6 +204,28 @@ public protocol ActionProtocol: BindingTargetProtocol {
 	/// `NoError` can be used.
 	associatedtype Error: Swift.Error
 
+	/// Initializes an action that will be conditionally enabled, and creates a
+	/// SignalProducer for each input.
+	///
+	/// - warning: This is an internal implementation detail of `Action.init(input:_:)`
+	///            and may be removed in a future version of ReactiveSwift.
+	///            Use `Action.init(enabledIf:_:)` or `Action.init(input:_:)` instead.
+	///
+	/// - note: Action guarantees that changes to `state` are observed in a
+	///         thread-safe way. Thus, the value passed to `isEnabled` will
+	///         always be identical to the value passed to `execute`, for each
+	///         application of the action.
+	///
+	/// - parameters:
+	///   - state: A property that provides the current state of the action
+	///            whenever `apply()` is called.
+	///   - enabledIf: A predicate that, given the current value of `state`,
+	///                returns whether the action should be enabled.
+	///   - execute: A closure that returns the signal producer returned by
+	///              calling `apply(Input)` on the action, optionally using
+	///              the current value of `state`.
+	init<State: PropertyProtocol>(_internalState property: State, enabledIf isEnabled: @escaping (State.Value) -> Bool, _ execute: @escaping (State.Value, Input) -> SignalProducer<Output, Error>)
```

Comment:
Sounds great. The underscore label of the designated initializer can be stripped now, can't it? It has a valid use case now as a public API (state with any arbitrary isEnabled predicate).